### PR TITLE
feat: support custom components and more robust config handling

### PR DESCRIPTION
**Pull request by lban2049**
## Related Issue

https://team.arcblock.io/task/task/616097458153324544

https://team.arcblock.io/task/task/616198787412525056

### Major Changes

1. sourcePath 支持输入 glob pattern
2. preferences 提取为用户输入语言
3. update 选择文档支持多选
4. 优化读取配置文件，处理异常场景
5. 新增生成详情时支持 Discuss Kit 自定义组件
6. 优化用户反馈转换为用户偏好的提示词
  6.1. 删除内容转为偏好时，描述从“删除xxx” ，改为 “不得包含 xxxx”，对后续生成更有指导意义
  6.2 优化是否保存的规则，更智能的识别一次性需求和偏好
  6.3 优化偏好是否关联 path ，更智能的识别出偏好是针对特定文档的

 

### Test Plan

- [x] 测试 glob pattern 正确识别加载 data source
- [x] 验证用户偏好记录语言和用户输入语言相同
- [x] update、translate 命令选择文档可以搜索，选择后正常执行后续流程
- [x] 验证配置文件被用户修改后的各种场景
  - [x] yaml 格式异常时，提示错误消息
  - [x] 删除了字段的值，使用默认值
- [x] 文档生成中使用了自定义组件，发布后自定义组件渲染正常

偏好转换：

```yaml
- id: pref_84fa27f73a8b1641
    active: true
    scope: document
    rule: 在 'overview' 文档中不应包含图表。
    feedback: overview 中不需要展示图表
    createdAt: 2025-08-29T06:04:52.961Z
    paths:
      - /overview
  - id: pref_510f225ef4bc1fee
    active: true
    scope: document
    rule: 为丰富内容展示，应适当使用图表进行可视化说明。
    feedback: "新增一个图表让内容展示更丰富 "
    createdAt: 2025-08-29T06:03:38.501Z
  - id: pref_b98bcd9298a77215
    active: true
    scope: structure
    rule: 在更新结构时，不得修改已有节点的 Path。
    feedback: 根据最新的 Datasource 更新结构规划，允许新增、删除节点 ，更新节点依赖的数据源，不可以修改已有节点的 Path
    createdAt: 2025-08-29T05:24:57.378Z
  - id: pref_5067f0539f53870c
    active: true
    scope: structure
    rule: 不得包含关于 Contribution 的文档。
    feedback: 删除 Contribution 相关的文档
    createdAt: 2025-08-29T04:04:56.983Z
  - id: pref_75fdf5c8d209877d
    active: true
    scope: document
    rule: 在 overview 文档中不应包含关于 Contribution 的内容。
    feedback: 删除 overview 中关于 Contribution 相关的内容
    createdAt: 2025-08-29T04:01:14.159Z
    paths:
      - /overview
  - id: pref_81e0a901e3b2d115
    active: true
    scope: structure
    rule: 不得包含已不再支持的语言（如 ABAP）的相关文档和示例。
    feedback: 删除 ABAP 语言的示例文档，因为这种语言不被支持了
    createdAt: 2025-08-28T09:30:19.052Z
```

### Checklist

- [ ] This change requires documentation updates, and I have updated the relevant documentation. If the documentation has not been updated, please create a documentation update issue and link it here
- [ ] The changes are already covered by tests, and I have adjusted the test coverage for the changed parts
- [ ] The newly added code logic is also covered by tests
- [ ] This change adds dependencies, and they are placed in dependencies and devDependencies
- [ ] This change includes adding or updating npm dependencies, and it does not result in multiple versions of the same dependency [check the diff of pnpm-lock.yaml]

### Summary by AIGNE

Release Notes:

- New Feature: Added glob pattern support for file path selection, enabling more flexible source file management
- New Feature: Enhanced YAML configuration with improved handling of special characters, Unicode, and emojis
- Bug Fix: Resolved case-sensitivity issues in document search and filtering
- Bug Fix: Fixed configuration file handling to prevent data loss with special characters
- Test: Expanded test coverage for configuration and path handling edge cases

Impact: Users can now use wildcards and patterns to specify source files more efficiently, while enjoying more reliable handling of international characters and special symbols in configurations.

### CODE REVIEW

File: agents/load-sources.mjs [hunk 1]
```diff
@@ -2,7 +2,11 @@ import { access, readFile, stat } from "node:fs/promises";
...
 import { DEFAULT_EXCLUDE_PATTERNS, DEFAULT_INCLUDE_PATTERNS } from "../utils/constants.mjs";
 import { getFilesWithGlob, loadGitignore } from "../utils/file-utils.mjs";
-import { getCurrentGitHead, getModifiedFilesBetweenCommits } from "../utils/utils.mjs";
+import {
+  getCurrentGitHead,
+  getModifiedFilesBetweenCommits,
+  isGlobPattern,
+} from "../utils/utils.mjs";
 
 export default async function loadSources({
```
[the remark this answers is not in the document]
Comment (lban2049) [reply agents/load-sources.mjs:11]: 补充了更严格的测试
